src/client/Microsoft.Identity.Client/AppConfig/CacheOptions.cs

```diff
@@ -50,5 +63,9 @@ public CacheOptions(bool useSharedCache)
         /// </remarks>
         public bool UseSharedCache { get; set; }
 
+        /// <summary>
+        /// 
+        /// </summary>
+        public IEnumerable<string> ExtraTokenParamsToCache { get; set; }
```

Comment:
I suggest you use a topic branch e.g. "travis/sti" and create small individual PRs for all of these improvements. ExtraTokenParams should be separated from JSON.